crates/consensus/src/transaction/eip4844.rs

```diff
@@ -3,13 +3,303 @@ use alloy_eips::{eip2930::AccessList, eip4844::DATA_GAS_PER_BLOB};
+        }
+    }
+
+    fn gas_limit(&self) -> u64 {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.gas_limit,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.gas_limit,
+        }
+    }
+
+    fn gas_price(&self) -> Option<U256> {
+        None
+    }
+
+    fn set_chain_id(&mut self, chain_id: ChainId) {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.chain_id = chain_id,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.chain_id = chain_id,
+        }
+    }
+
+    fn input(&self) -> &[u8] {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.input.as_ref(),
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.input.as_ref(),
+        }
+    }
+
+    fn input_mut(&mut self) -> &mut Bytes {
+        match self {
+            TxEip4844::TxEip4844(tx) => &mut tx.input,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => &mut tx.input,
+        }
+    }
+
+    fn nonce(&self) -> u64 {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.nonce,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.nonce,
+        }
+    }
+
+    fn set_gas_limit(&mut self, limit: u64) {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.gas_limit = limit,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.gas_limit = limit,
+        }
+    }
+
+    fn set_gas_price(&mut self, price: U256) {
+        let _ = price;
+    }
+
+    fn set_input(&mut self, data: Bytes) {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.input = data,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.input = data,
+        }
+    }
+
+    fn set_nonce(&mut self, nonce: u64) {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.nonce = nonce,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.nonce = nonce,
+        }
+    }
+
+    fn set_to(&mut self, to: TxKind) {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.to = to,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.to = to,
+        }
+    }
+
+    fn set_value(&mut self, value: U256) {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.value = value,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.value = value,
+        }
+    }
+
+    fn to(&self) -> TxKind {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.to,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.to,
+        }
+    }
+
+    fn value(&self) -> U256 {
+        match self {
+            TxEip4844::TxEip4844(tx) => tx.value,
+            TxEip4844::TxEip4844WithSidecar((tx, _)) => tx.value,
+        }
+    }
+}
+
+/// [EIP-4844 Blob Transaction](https://eips.ethereum.org/EIPS/eip-4844#blob-transaction)
+///
+/// A transaction with blob hashes and max blob fee. It does not have the Blob sidecar.
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Default)]
-pub struct TxEip4844 {
+pub struct BlobTx {
```

Comment:
this type should be called TxEip4844

Reply:
Yep sg. Not married to names personally—what should we call the enum?